modules/heat_conduction/src/kernels/JouleHeatingSource.C

```diff
@@ -0,0 +1,53 @@
+#include "JouleHeatingSource.h"
+
+template<>
+InputParameters validParams<JouleHeatingSource>()
+{
+  InputParameters params = validParams<HeatSource>();
+  params.addCoupledVar("elec", "Electric potential for joule heating.");
+  params.addCoupledVar("args", "Vector of arguments of the diffusivity");
+  params.addParam<MaterialPropertyName>("electrical_conductivity", "electrical_conductivity", "material property providing electrical resistivity of the material.");
```

Comment:
Calculating electrical conductivity in the material helps in using MatDiffusion kernel directly without parsed functions for the electric potential variable. May be I can skip storing resistivity as material property.